pixeltable/exec/expr_eval/expr_eval_node.py

```
@@ -390,6 +390,10 @@ def _done_cb(self, t: asyncio.Task) -> None:
         # end the main loop if we had an unhandled exception
         try:
             t.result()
+        except KeyboardInterrupt:
```

Doesn't this end up swallowing the exception?

---

added comment there explaining why we dont record keyboard interrupt

---

KeyboardInterrupt is initiated by the OS, which sends a SIGINT signal to the Python process's main thread. In response, the main thread's event loop begins a shutdown sequence, canceling all awaiting or pending tasks.

Any task that is running at the same time and hasn't yielded control back to the event loop won't see this cancellation immediately. Instead, it might be directly interrupted by the interpreter's signal handler, which raises a KeyboardInterrupt in the task's context. If the task doesn't handle this exception, its result is marked with the KeyboardInterrupt.

When the task's completion callback is finally triggered, likely in the future during a subsequent operation, it will raise this stored KeyboardInterrupt. This is why you need to absorb it: it's a "phantom" exception from an already-handled user cancellation, and it shouldn't be treated as a new, unhandled error.